netmap/netmap_contract.go

```diff
@@ -259,28 +287,63 @@ func AddPeer(nodeInfo []byte) {
 		common.RemoveVotes(ctx, id)
 	}
 
-	addToNetmap(ctx, candidate)
-	runtime.Notify("AddPeerSuccess", interop.PublicKey(publicKey))
+	addToNetmap(ctx, publicKey, candidate)
+}
+
+// updates state of the network map candidate by its public key in the contract
+// storage, and throws UpdateStateSuccess notification after this.
+//
+// State MUST be from the NodeState enum.
+func updateCandidateState(ctx storage.Context, publicKey interop.PublicKey, state NodeState) {
+	switch state {
+	case NodeStateOffline:
+		removeFromNetmap(ctx, publicKey)
+		runtime.Log("remove storage node from the network map")
```

Comment:
Why do you need these logs? There's a notification in the end of `updateCandidateState`, it should be enough.

Reply:
Maybe to distingush notification losses from non-throws. I don't mind to have such log messages.